src/redisearch_rs/rlookup/src/lib.rs

```diff
@@ -18,3 +18,33 @@ pub use lookup::{
     RLookup, RLookupKey, RLookupKeyFlag, RLookupKeyFlags, RLookupOption, RLookupOptions,
 };
 pub use row::RLookupRow;
+use value::RSValueTrait;
+
+/// Retrieves an item from the given `RLookupRow` based on the provided `RLookupKey`.
+/// The function first checks for dynamic values, and if not found, it checks the sorting vector
+/// if the `SvSrc` flag is set in the key.
+/// If the item is not found in either location, it returns `None`.
+///
+/// # Lifetime
+/// The returned reference is tied to the lifetime of the input `RLookupRow`.
+pub fn rlookup_get_item<'a>(
+    key: &RLookupKey,
+    row: &'a RLookupRow<impl RSValueTrait>,
+) -> Option<&'a impl RSValueTrait> {
+    // 1. Check dynamic values first
+    if row.len() > key.dstidx as usize {
+        return row.dyn_values()[key.dstidx as usize].as_ref();
+    }
+
+    if !key.flags.contains(RLookupKeyFlag::SvSrc) {
+        return None;
+    }
+
+    // 3. Linearly search the sorting vector if it exists
+    let sv = row.sorting_vector()?;
+    if sv.len() <= key.svidx as usize {
+        return None;
+    }
+
+    Some(&sv[key.svidx as usize])
```

Comment:
instead of the bounds-checking and early return, can we use `.get()` on the sorting vector to get an `Option<RSValue>`?

instead of:
```rust
if sv.len() <= key.svidx as usize {
        return None;
}

Some(&sv[key.svidx as usize])
```

do this?
```rust
sv.get(key.svidx)
```